bin/git-utimes

```diff
@@ -2,20 +2,28 @@
 #
 # Change files modification time to their last commit date
 #
-if [ "$1" = --touch ]; then
+if [ "$1" = "--touch" ]; then
   # Internal use option only just to parallelize things.
-  shift
+  newer_flag=""
+  [ "$2" = "--newer" ] && newer_flag="true"
+  shift 2
+  bsd=$(date -j > /dev/null 2>&1 && echo 'true')
+  if [ -n "$bsd" ]; then
+    stat_flags="-f %m"
+    date_flags="-r"
+  else
+    stat_flags="-c %Y"
+    date_flags="-d@"
+  fi
   for f; do
-    iso_t=$(git --no-pager log --no-renames --pretty=format:%cI -1 @ -- "$f" 2>/dev/null)
-    if [ -n "$iso_t" ]; then
-      bsd=$(date -j > /dev/null 2>&1 && echo 'true')
-      if [ "$bsd" == "true" ]; then
-        t=$(date -j -f %Y-%m-%dT%H:%M:%S "$iso_t" +%Y%m%d%H%M.%S)
-      else
-        t=$(date -d"$iso_t" +%Y%m%d%H%M.%S)
+    git_s=$(git --no-pager log --no-renames --pretty=format:%ct -1 @ -- "$f" 2>/dev/null)
+    mod_s=$(stat $stat_flags "$f" 2>/dev/null)
+    if [ -n "$git_s" ] && [ -n "$mod_s" ] && [ $mod_s -ne $git_s ]; then
```

Comment:
Would be better to double-quote every var

Reply:
git_s and mod_s are integers but OK, done.